### DISCUSSION TITLE
What is the supposed filesize of entry.js?

### DISCUSSION THREAD

**Discussion by MarcoTroost** [nuxt/nuxt#23729] [Questions] [answered]
Hi!

Our team is exploring the possiblities of migrating from a native vue SPA to a Nuxt SSR solution. 

Following the basic installation using `npx nuxi@latest init my-app`, followed by `npm run build`, the result is an entry.js file of nearly 200 kB. 

<img width="801" alt="Scherm­afbeelding 2023-10-17 om 19 53 56" src="https://github.com/nuxt/nuxt/assets/16369103/cac2a7d4-f77b-4a79-be1f-60ca01d6a6f7">

This is just the basic scaffolding. We haven't even created something!
I'm sad to say that this enormous filesize is seriously underwhelming. The argument for Nuxt is not only ease of use, but also speed & performance.

Are we doing something wrong? 

kind regards, Marco

**Comment by warflash** [comment]
https://stackblitz.com/edit/github-duzxuv?file=app.vue building this I get the following output:
![image](https://github.com/nuxt/nuxt/assets/22006082/83894c65-c05f-4e0c-979a-1d54b08de23f)
I'd assume you kept the nuxtwelcome component included in the build.

**Reply by MarcoTroost** [reply]
Hi @warflash thnx, i'll give it a try. 

Still too large imho. 103 kB for just an empty page? The bare html equivalent probably isn't even 1 kB. 

Where does all this overhead come from? 

I very much appreciate the ease of development Nuxt provides and I really do not mean to be rude. With pagespeed being so important and all, this overhead really works against Nuxt.

kind regards, Marco

**Reply by TheAlexLichter** [reply, marked as answer]
> Still too large imho. 103 kB for just an empty page? The bare html equivalent probably isn't even 1 kB.

This is comparing apples to pears.

First, the gzipped + minified size is ~40kB (every browser supports at least GZIP, even better [Brotli](https://www.brotli.pro/).
Second, you have lots of functionality (including Vue and Nuxt) in these 50kB. With a blank HTML page, you have just that - nothing else.

> Where does all this overhead come from?

Use `nuxi analyze` to see what your build consists of ☺️ For an empty project this will be mainly:

* Vue's dom runtime (~2 kB)
* Vue's core runtime (~18.75 kB)
* Vue's reactivity (~ 4 kB)
* unhead (for seo/meta) (~2 kB)
* Further UnJS + Nuxt.js libraries (such as ufo, defu, ofetch, ...)
* Nuxt app (~8kB)
* The welcome component (which will be removed in your actual app)

![image](https://github.com/nuxt/nuxt/assets/640208/44882bef-7705-484f-a7af-feb30f446474)

Without the `welcome` component you see the following:

![image](https://github.com/nuxt/nuxt/assets/640208/f5556a1f-9138-4747-ad7c-d497b52e1000)

---

Be aware that these 50kB are most likely smaller than most images, third party libraries or often even own code you use on your website and are important for the baseline functionalities, such as providing Vue and Nuxt, error handling, Nuxt hooks, dealing with `meta` tags (etc. etc.).

---

We also have [tests in place](https://github.com/nuxt/nuxt/blob/main/test/bundle.test.ts) to ensure the initial JavaScript is as minimal as possible (for a minimal application) to avoid bloat ☺️ 

---

If you **don't need JS "at all"**, you can fully remove scripts in Nuxt, leaving you with no overhead, but also no SPA capabilities.

**Reply by ishaanparlikar** [reply]
is this also goes into the bundle when creating pre-render or SSG build as well? and is there any way to reduce it down like features we dont need?

**Reply by TheAlexLichter** [reply]
> is this also goes into the bundle when creating pre-render or SSG build as well? and is there any way to reduce it down like features we dont need?

Yes. You can remove *all* JS if you want (experimental.noScripts) or remove the router by not using the pages/ folder.

**Reply by ishaanparlikar** [reply]
But then if we remove pages folder build command will crawl through only app.vue right? Will not generate other pages?

**Reply by TheAlexLichter** [reply]
Yes - if you have multiple pages then the router would be helpful 😋

**Reply by ishaanparlikar** [reply]
Ahh. Got it. Is there another way? 
We really want nuxt to work out for us.

**Reply by cjpearson** [reply]
I've also been looking into the bundle size of the default application. When I run `nuxi analyze` it shows the vue packages taking up 29.8kb gzipped. According to the Vue docs, that's the maximum size. https://vuejs.org/about/faq#is-vue-lightweight Since the default application isn't using very many features is it possible that tree-shaking Vue is not working?

**Reply by TheAlexLichter** [reply]
@cjpearson Interesting result! Might be worth raising an issue for it 👍

**Reply by cjpearson** [reply]
Done! https://github.com/nuxt/nuxt/issues/28808

**Comment by MarcoTroost** [comment]
Hi @manniL Thank you for your elaborate answer. This certainly helps me better understand the underlying logic & provides me with the proper arguments to make the case for using Nuxt in our organisation.

In fact, i think that your explanation could prove beneficial to be incorporated on the Nuxt website itself. 
One does really get a lot of value per kB.

kind regards, Marco

**Reply by TheAlexLichter** [reply]
@MarcoTroost No problem at all ☺️ 

I agree with that. If you want to, please raise an issue and if you feel like - happy to see a PR with some text suggestions (draft is totally fine) for https://github.com/nuxt/nuxt.com
